processing/src/main/java/io/druid/query/search/search/SearchQueryConfig.java

```diff
@@ -27,12 +27,32 @@
  */
 public class SearchQueryConfig
 {
+  public static final String CTX_KEY_STRATEGY = "searchStrategy";
+
   @JsonProperty
   @Min(1)
   private int maxSearchLimit = 1000;
 
+  @JsonProperty
+  private String searchStrategy = AutoStrategy.NAME;
```

Comment:
Sounds good. Let's make index-only strategy default. We can change later if auto strategy becomes sufficiently smart and efficient.